geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java

```diff
@@ -64,52 +65,60 @@ public PropertyManagement(
 
 	public Object get(
 			final ParameterEnum propertyName ) {
-		return properties.get(toPropertyName(propertyName));
+			return properties.get(toPropertyName(propertyName));
 	}
 
 	public void store(
 			final ParameterEnum propertyName,
 			final Object value ) {
-		properties.put(
-				toPropertyName(propertyName),
-				value);
-
+		synchronized (MUTEX) {
+			properties.put(
+					toPropertyName(propertyName), value);
+		}
 	}
 
 	public Object storeIfEmpty(
 			final ParameterEnum propertyName,
 			final Object value ) {
 		final String pName = toPropertyName(propertyName);
-		if (!properties.containsKey(pName)) {
-			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
-			properties.put(
-					pName,
-					value);
-			return value;
-		}
-		return properties.get(pName);
+		synchronized (MUTEX) {
```

Comment:
@rwgdrummer 

All the synchronization here was because there are a few places that have the  
if (!contains) then put
pattern going on.  Definitely not an atomic operation, and it looked like this might be called from multiple threads (just at a quick glance though) - so put a lock around the pattern above + any place that was calling adding items.

Probably can be removed/unnecessary if we want to mark everything as not threadsafe. (depends on use requirements)